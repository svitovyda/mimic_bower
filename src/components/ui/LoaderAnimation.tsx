import { LoaderContainer, LoaderRing, LoaderRingInner, LoaderTextContainer } from "../../styles/ui/LoaderAnimation";
import * as React from "react";

export const LoaderAnimation: React.FC = React.memo(() => {
  return (
    <LoaderContainer>
      <LoaderRing />
      <LoaderRingInner />
      <LoaderTextContainer>loading</LoaderTextContainer>
    </LoaderContainer>
  );
});

LoaderAnimation.displayName = "LoaderAnimation";
