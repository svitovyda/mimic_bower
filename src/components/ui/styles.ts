import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  html {
    font-family:sans-serif;
    -ms-text-size-adjust:100%;
    -webkit-text-size-adjust:100%
  }
  body {
    font: 16px "Helvetica Neue",Helvetica,Arial,sans-serif;
    line-height: 1.5;
    padding: 0;
    margin: 0;
    color: #543729
  }
`;
