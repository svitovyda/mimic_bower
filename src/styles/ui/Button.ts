import styled from 'styled-components';

export const Button = styled.button`
  cursor: pointer;
  outline: 0;
  display: inline-block;
  font-weight: bold;
  line-height: 1.5;
  height: 45px;
  text-align: center;
  background-color: white;
  border: 1px solid #ccc;
  border-radius: 2px;
  padding: 6px 12px;
  font-size: 18px;
  color: #00acee;
  :hover:enabled {
    border-color: #ef5734;
  }
  :disabled {
    color: lightgray;
    border-color: lightgray;
  }
`;
