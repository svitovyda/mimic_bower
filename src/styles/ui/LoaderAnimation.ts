import styled, { keyframes } from "styled-components";

const RotateRing = keyframes`
  0% { transform: rotate(0) }
  100% { transform: rotate(360deg) }
`;

const RotateInner = keyframes`
  0% { transform: rotate(0) }
  100% { transform: rotate(-360deg) }
`;

const TextLoading = keyframes`
  0% { opacity: 0 }
  20% { opacity: 0 }
  50% { opacity: 1 }
  100% { opacity: 0 }
`;

export const LoaderContainer = styled.div`
  height: 50px;
  width: 50px;
  margin: 5px auto;
`;

export const LoaderRing = styled.div`
  position: relative;
  top: 0;
  left: 0;
  height: 50px;
  width: 50px;
  border-radius: 100%;
  border: 3px solid transparent;
  border-color: transparent lightgrey transparent lightgrey;
  animation: ${RotateRing} 2s linear 0s infinite normal;
  transform-origin: 50% 50%;
`;

export const LoaderRingInner = styled.div`
  position: relative;
  top: -52px;
  left: 5px;
  width: 43px;
  height: 43px;
  border-radius: 100%;
  border: 2px solid transparent;
  border-color: transparent lightgrey transparent lightgrey;
  animation: ${RotateInner} 1s linear 0s infinite normal;
  transform-origin: 50% 50%;
`;

export const LoaderTextContainer = styled.div`
  position: relative;
  top: -80px;
  left: -22px;
  width: 100px;
  user-select: none;
  animation: ${TextLoading} 3s linear 0s infinite normal;
  color: lightgrey;
  font:
    bold 8px "Helvetica Neue",
    Helvetica,
    Arial,
    sans-serif;
  opacity: 0;
  text-align: center;
  text-transform: uppercase;
`;
