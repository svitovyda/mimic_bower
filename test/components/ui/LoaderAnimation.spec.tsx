import { LoaderAnimation } from '../../../src/components/ui/LoaderAnimation';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import * as React from 'react';

describe('LoaderAnimation', () => {
  it('renders without error', () => {
    const component = render(<LoaderAnimation />);

    expect(component).toMatchSnapshot();
  });
});
