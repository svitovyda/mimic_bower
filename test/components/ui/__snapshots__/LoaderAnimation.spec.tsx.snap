// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderAnimation renders without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  height: 50px;
  width: 50px;
  margin: 5px auto;
}

.c1 {
  position: relative;
  top: 0;
  left: 0;
  height: 50px;
  width: 50px;
  border-radius: 100%;
  border: 3px solid transparent;
  border-color: transparent lightgrey transparent lightgrey;
  -webkit-animation: LtRVX 2s linear 0s infinite normal;
  animation: LtRVX 2s linear 0s infinite normal;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
}

.c2 {
  position: relative;
  top: -52px;
  left: 5px;
  width: 43px;
  height: 43px;
  border-radius: 100%;
  border: 2px solid transparent;
  border-color: transparent lightgrey transparent lightgrey;
  -webkit-animation: cLNnjC 1s linear 0s infinite normal;
  animation: cLNnjC 1s linear 0s infinite normal;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
}

.c3 {
  position: relative;
  top: -80px;
  left: -22px;
  width: 100px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-animation: kRlIGi 3s linear 0s infinite normal;
  animation: kRlIGi 3s linear 0s infinite normal;
  color: lightgrey;
  font: bold 8px "Helvetica Neue", Helvetica, Arial, sans-serif;
  opacity: 0;
  text-align: center;
  text-transform: uppercase;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        />
        <div
          class="c2"
        />
        <div
          class="c3"
        >
          loading
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  height: 50px;
  width: 50px;
  margin: 5px auto;
}

.c1 {
  position: relative;
  top: 0;
  left: 0;
  height: 50px;
  width: 50px;
  border-radius: 100%;
  border: 3px solid transparent;
  border-color: transparent lightgrey transparent lightgrey;
  -webkit-animation: LtRVX 2s linear 0s infinite normal;
  animation: LtRVX 2s linear 0s infinite normal;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
}

.c2 {
  position: relative;
  top: -52px;
  left: 5px;
  width: 43px;
  height: 43px;
  border-radius: 100%;
  border: 2px solid transparent;
  border-color: transparent lightgrey transparent lightgrey;
  -webkit-animation: cLNnjC 1s linear 0s infinite normal;
  animation: cLNnjC 1s linear 0s infinite normal;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
}

.c3 {
  position: relative;
  top: -80px;
  left: -22px;
  width: 100px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-animation: kRlIGi 3s linear 0s infinite normal;
  animation: kRlIGi 3s linear 0s infinite normal;
  color: lightgrey;
  font: bold 8px "Helvetica Neue", Helvetica, Arial, sans-serif;
  opacity: 0;
  text-align: center;
  text-transform: uppercase;
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      />
      <div
        class="c2"
      />
      <div
        class="c3"
      >
        loading
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
